endpoint/endpoint.go

```diff
@@ -343,3 +343,22 @@ type DNSEndpointList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []DNSEndpoint `json:"items"`
 }
+
+// Apply filter based on EndpointKey (using DNSName, RecordType & SetIdentifier)
```

Comment:
Nitpick based on https://go.dev/doc/comment (though much of this project doesn't seem to follow this guidance anyway...): the doc comment should "start with [a] complete sentence naming the declared symbol" and "should not explain internal details".

Maybe something like the following.

```suggestion
// RemoveDuplicates returns a slice holding the unique endpoints.
```